### PR TITLE
move ingest & query to top-level client

### PR DESCRIPTION
**Pull request by schehata**

### CODE REVIEW

File: lib/client.ts [hunk 1]
```diff
@@ -1,13 +1,303 @@
...
+import HTTPClient, { ClientOptions } from './httpClient';
+import { gzip } from 'zlib';
+import { promisify } from 'util';
+import { Readable, Stream } from 'stream';
 
-export default class Client {
+export default class Client extends HTTPClient {
     datasets: datasets.Service;
     users: users.Service;
+    localPath = '/api/v1/datasets';
```
Comment (bahlo) [inline lib/client.ts:11]: Should this be `/api/v1` because it's not a specific client?

File: lib/client.ts [hunk 2]
```diff
@@ -1,13 +1,303 @@
...
+            })
+            .then((response) => {
+                return response.data;
+            });
+    };
+
+    aplQuery = (apl: string, options?: QueryOptions): Promise<QueryResult> => this.query(apl, options);
+}
+
+export enum ContentType {
```
Comment (bahlo) [inline lib/client.ts:99]: I think we can keep the models in the datasets file, wdyt?

Comment (schehata) [reply lib/client.ts:99]: hmm I am not sure about that, I feel its a little bit weird to use `client.ingest` and then use params from a different namespace. how is it in the rs sdk?

File: lib/client.ts (hunk 1 above)
Comment (schehata) [reply lib/client.ts]: updated

File: lib/client.ts (hunk 2 above)
Comment (schehata) [reply lib/client.ts:99]: after our talk, I believe this one is ready to be merged.

**Pull request merged**